### PR TITLE
Added label-based mechanism to identify docker-gen container. This is needed w…

### PR DESCRIPTION
**Pull request by chicco785**
Added label-based mechanism to identify docker-gen container. This is needed when using docker compose or other container mechanisms that define dynamically container's name.

### CODE REVIEW

File: app/entrypoint.sh [hunk 1]
```diff
@@ -79,6 +92,7 @@ source /app/functions.sh
 
 if [[ "$*" == "/bin/bash /app/start.sh" ]]; then
     check_docker_socket
+    get_nginx_gen_cid
```
Comment (helderco) [inline app/entrypoint.sh]: I think this should be: 
```
[[ -z "${NGINX_DOCKER_GEN_CONTAINER:-}" ]] && get_nginx_gen_cid
``` 
to allow still using a manual environment variable when you do have access to a static name.

File: app/entrypoint.sh [hunk 2]
```diff
@@ -45,6 +45,19 @@ function get_nginx_proxy_cid {
     fi
 }
 
+function get_nginx_gen_cid {
+    # Check if any container has been labelled as the nginx gen container.
+    local labeled_cid=$(docker_api "/containers/json" | jq -r '.[] | select( .Labels["com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_gen"] == "true")|.Id')
+    if [[ ! -z "${labeled_cid:-}" ]]; then
+        export NGINX_DOCKER_GEN_CONTAINER=$labeled_cid
+    fi
+    if [[ -z "${NGINX_DOCKER_GEN_CONTAINER:-}" ]]; then
```
Comment (helderco) [inline app/entrypoint.sh]: Remove this condition to make it optional, don't error out. Document it instead.

File: app/nginx_location.conf [hunk 3]
```diff
@@ -1,4 +1,5 @@
-location ^~ /.well-known/acme-challenge/ {
+location ~ /.well-known {
```
Comment (helderco) [inline app/nginx_location.conf]: Out of scope, changes should be discarded.

**Comment by chicco785** [comment]
the change to `app/nginx_location.conf `may not be required. it's the results of different attempts to fix the issues i was facing in getting the certificate created. but the problem indeed was all related to the impossibility of connecting to the correct container due to the dynamic naming. the `com.github.jrcs.letsencrypt_nginx_proxy_companion.nginx_gen` label can be optional of course. let me know what you think.

**Comment by helderco** [comment]
I need this, if you hadn't, I would! 👍

**Comment by helderco** [comment]
I need this, if you hadn't, I would! 👍

**Comment by helderco** [comment]
I found an issue with containers getting redeployed during letsencrypt lifecycle, so I made my own attempt with #231. It's working in production right now, with the [helder/letsencrypt-nginx-proxy-companion](https://hub.docker.com/r/helder/letsencrypt-nginx-proxy-companion/) fork (temporary until the PR is merged).

**Comment by helderco** [comment]
#231 was merged so this can be closed.

**Comment by JrCs** [comment]
Closed by #231

**Pull request closed without merging**